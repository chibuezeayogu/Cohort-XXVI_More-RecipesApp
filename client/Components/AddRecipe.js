import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Input, Icon, Button, Col, Table} from 'react-materialize';
import axios from 'axios';
import Preloader from './Preloder';
//require('dotenv').config();


const CLOUDINARY_URL = process.env.CLOUDINARY_URL;
const CLOUDINARY_UPLOAD_PRESET = process.env.CLOUDINARY_UPLOAD_PRESET;


class AddRecipe extends Component{

    constructor(props) {
        super(props);
        this.state = {
            file: '',
            imagePreviewUrl: '', 
            ImageURL: '',
            arrIngredients:[],
            arrProcedures:[]
        };
      }

      componentWillMount(){
        // if(!this.props.userData.isAuthenticated){
        //     this.context.router.push('/sign');
        // }
        const token = localStorage.getItem('jwtToken');
        if(!token){
            this.context.router.push('/signin');
        }
    }
    //   componentWillUpdate(nextProps, nextState) {
    //     console.log(nextState.arrIngredients);
    // }
   
      _addIngredients(e){
          e.preventDefault();
          const ingredients = document.getElementById('ingredients').value;
          if(!ingredients){
                alert("Ingredients field is empty");
          }else {

            const newIngredients = this.state.arrIngredients.concat(ingredients);
                this.setState({
                    arrIngredients: newIngredients
                });
            }
      }

     



      _addProcedures(e){
        e.preventDefault();
        const procedures = document.getElementById('procedures').value;
        if(!procedures){
              alert("Ingredients field is empty");
        }else {

          const newProcedures = this.state.arrProcedures.concat(procedures);
              this.setState({
                    arrProcedures: newProcedures
              })
          }
        
          this.ingredient.state.reset();
      }

      _handleImageChange(e) {
        e.preventDefault();
    
        let reader = new FileReader();
        let file = e.target.files[0];
        const formData = new FormData;
        formData.append('file', file);
        formData.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);
        
        // if (!file.type.match('image.*')){

        // }else{

        // }
        
        $.ajax({
            type: "POST",
            url: CLOUDINARY_URL,
            dataType: 'json',
            data: formData,
            contentType:false,
            processData: false,
            success: ((data) => {
                this.setState({ImageURL:data.secure_url});
            })
        });

        reader.onloadend = () => {
          this.setState({
            file: file,
            imagePreviewUrl: reader.result
          });
        };
    
        reader.readAsDataURL(file);
      }
      
    handleOnsubmit(e) {
        e.preventDefault();
        const title = this.title.state.value;
        const description = this.description.state.value;
        const ingredients = this.ingredients.state.value;
        const procedures = this.procedures.state.value;
        const imageURL = this.state.ImageURL;
        this.props.addRecipe(title, description, ingredients, procedures, imageURL);
        this.refs.AddRecipeForm.reset();
        this.setState({
            imagePreviewUrl:''
        });
    }
    render() {
        // let {imagePreviewUrl} = this.state;
        // let $imagePreview = null;
        // if (imagePreviewUrl) {
        //   $imagePreview =(<img  class="responsive-img circle" src={imagePreviewUrl}/>);
        // } else {
        //   $imagePreview = " ";
        // }
       const  { arrIngredients, arrProcedures } = this.state;
        return (
            <div className="container"> 
                <h1 className="center">Add Recipe</h1>
                <hr/>
                <form encType="multipart/form-data" methos="post" className="col l6 m8 s12 offset-l8 offset-m6" onSubmit={(e) => this.handleOnsubmit(e)} ref="AddRecipeForm">
                    <Row>
                        <Input s={12} label="Title"  type='text' ref={(input)=> {this.title=input}}><Icon>short_text</Icon></Input>
                    </Row>
                    <Row s={8}>
                        <Input s={12} label="Description" className='validate' type='text' ref={(input) => {this.description = input}}><Icon>short_text</Icon></Input>
                    </Row>
                    <Row>
                        <Input s={12} label="Ingredients" id="ingredients" type='text' ref={(input) => {this.ingredients = input}}><Icon>list</Icon></Input>
                        <Button waves='light' className='right' onClick={(e)=> {this._addIngredients(e)}}>Add Ingredient</Button>
                    </Row>
                    <Row>
                        <Table className="striped">
                            <tbody>
                                {arrIngredients
                                    .map((ingredient, i) => 
                                        <tr>
                                            <td className="s10">
                                                <input id="ingredients" type='text' value={ingredient}></input>
                                            </td>
                                            <td className="s1 right"><Icon>delete</Icon></td>
                                            <td className="s1 right"><Icon>edit</Icon></td>
                                        </tr>
                                    )
                                }
                            </tbody>
                        </Table>
                    </Row>
                    <Row>
                        <Input s={12} label="Procedures" id="procedures" type='text' ref={(input) => {this.procedures = input}}><Icon>list</Icon></Input>
                        <Button waves='light' className='right' onClick={(e)=> {this._addProcedures(e)}}>Add Procedures</Button>
                    </Row>
                    <Row>
                        <Table className="striped">
                            <tbody>
                                {arrProcedures
                                    .map((procedures, i) => 
                                        <tr>
                                            <td className="s10">{procedures}</td>
                                            <td className="s1 right"><Icon>delete</Icon></td>
                                            <td className="s1 right"><Icon>edit</Icon></td>
                                        </tr>
                                    )
                                }
                            </tbody>
                        </Table>
                    </Row>
                    <Row>
                        <Input s={6} id="recipe"  name="file" type='file' ref={(input) => {this.file = input}}   onChange={(e)=>this._handleImageChange(e)}><Icon>insert_photo</Icon></Input>
                        <Col s={6} ><img className="responsive-img img-wh z-depth-3" src={this.state.imagePreviewUrl && this.state.ImageURL?this.state.imagePreviewUrl:"select image to preview"}/></Col>
                    </Row>
                    <br />
                    <Row>
                        <Button waves='light' className='right'><Icon right>send</Icon>Add</Button>
                    </Row>
                </form>
            </div>
        );
    }
}

AddRecipe.contextTypes = {
    router: React.PropTypes.object
};


AddRecipe.propTypes = {
handleOnSubmit: PropTypes.func.isRequired,
_addIngredients:  PropTypes.func.isRequired,
_addProcedures:  PropTypes.func.isRequired,
_handleImageChange:  PropTypes.func.isRequired,
addRecipe: PropTypes.func.isRequired,
};

export default AddRecipe;