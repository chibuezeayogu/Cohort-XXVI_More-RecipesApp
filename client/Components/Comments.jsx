import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

const Comments = props => (
  <li className="collection-item avatar z-depth-1">
    <img
      src={props.review.User.imageUrl}
      alt="profile"
      className="circle responsive-img"
    />
    <span className="text-title bold">{props.review.User.firstName}</span>
    <p
      style={{ wordWrap: 'break-word' }}
    >
      {props.review.comment}
    </p>
    <div className="secondary-content">
      { moment(new Date(props.review.createdAt)).fromNow()}
    </div>
  </li>
);

Comments.propTypes = {
  review: PropTypes.shape({
    comment: PropTypes.string,
    createdAt: PropTypes.string,
    User: PropTypes.shape({
      firstName: PropTypes.string,
      imageUrl: PropTypes.string
    })
  }).isRequired
};

export default Comments;
