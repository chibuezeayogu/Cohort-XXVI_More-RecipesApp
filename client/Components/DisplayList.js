import React, { Component } from 'react';
import { Link } from 'react-router';

const Displaylist = React.createClass ({
    render(){
        const { recipes } = this.props;
        return(
            <td>
                list ingradients
            </td>
        );
    }
});

export default Displaylist;