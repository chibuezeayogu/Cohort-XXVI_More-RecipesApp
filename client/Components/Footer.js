import React, { Component }from 'react';
import PropTypes from 'prop-types';
import { Footer } from 'react-materialize';


class BottomFooter extends  Component {
    render() {
        return (
            <div>
                <Footer copyrights="&copy 2017 Copyright Text"
                moreLinks={
                    <a className="grey-text text-lighten-4 right" href="#!">Andela Project by Ayogu Chibueze></a>
                }
                
                className='black'
            >
                    
                </Footer>;
            </div>
        );
    }
}

BottomFooter.propTypes = {
    logout: PropTypes.func.isRequired,
    history: PropTypes.func.isRequired,
    push: PropTypes.func.isRequired,
    className: PropTypes.string,
    copyrights: PropTypes.string,
};

export default BottomFooter;