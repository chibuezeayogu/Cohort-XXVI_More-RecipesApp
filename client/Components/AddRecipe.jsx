require('dotenv').config();

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import { connect } from 'react-redux';
import { createRecipe } from '../action/actionCreators';
import SmallPreloader from './SmallPreloader.jsx';
import Footer from './Footer/Footer.jsx';
import UserMenu from './Header/UserMenu.jsx';
import { validateAddRecipe } from '../middleware/validateInputs';

const CLOUDINARY_URL = process.env.CLOUDINARY_URL;
const CLOUDINARY_UPLOAD_PRESET = process.env.CLOUDINARY_UPLOAD_PRESET;

/**
 * @description AddRecipe component for user to create a new recipe
 *
 * @class
 *
 * @extends Component
 *
 */
class AddRecipe extends Component {
   /**
   * @description initialize state and binds functiom
   *
   * @constructor
   *
   * @memberOf AddRecipe
   *
   * @returns {void}
   *
   */
  constructor() {
    super();
    this.state = {
      title: '',
      description: '',
      ingredients: '',
      procedures: '',
      imageUrl: '',
      image: {},
      errors: {},
      disabled: false,
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleOnsubmit = this.handleOnsubmit.bind(this);
  }

   /**
   * @description routes to /recipes if user is authenticated
   *
   * @method
   *
   * @memberOf AddRecipe
   *
   * @returns {void}
   *
   */
  componentWillMount() {
    const token = localStorage.getItem('jwtToken');
    if (!token) {
      this.props.history.push('/signin');
    }
  }
   /**
   * @description routes to /recipes if user is authenticated
   *
   * @method
   *
   * @memberOf AddRecipe
   *
   * @param {Object} nextProps
   *
   * @returns {void}
   *
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.recipeReducer.isCreated) {
      this.props.history.push('/recipes');
    }
  }


  handleChange(e) {
    e.preventDefault();
    this.setState({ errors: {} });
    this.setState({ [e.target.name]: e.target.value });
  }

  /**
   * @description get image input and add to state
   *
   * @method
   *
   * @memberOf AddRecipe
   *
   * @param {Event} e
   *
   * @returns {void}
   */
  handleImageChange(e) {
    e.preventDefault();
    const file = e.target.files[0];
    this.setState({ errors: {}, image: {} });
    this.setState({ image: file });
  }

  /**
   *
   * @description submits form
   *
   * @method
   *
   * @memberOf AddRecipe
   *
   * @param {Event} e
   *
   * @returns {void}
   *
   */
  handleOnsubmit(e) {
    e.preventDefault();
    const err = validateAddRecipe(this.state);
    if (err.isError) {
      return this.setState({ errors: err.errors });
    }

    this.setState({ disabled: true });
    const formData = new FormData();
    formData.append('file', this.state.image);
    formData.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);

    axios({
      url: CLOUDINARY_URL,
      method: 'POST',
      data: formData,
    }).then((data) => {
      this.setState({ imageUrl: data.data.secure_url, disabled: false });
      const {
        title, description, ingredients, procedures, imageUrl,
      } = this.state;
      console.log(imageUrl, 'image');
      this.props.createRecipe(title, description, ingredients, procedures, imageUrl);
    }).catch(() => { 
      this.setState({ disabled: false });
      Materialize.toast("Error! Please try again", 4000, 'red');
    });
  }

  /**
   *
   * @description renders JSX element
   *
   * @method
   *
   * @memberOf SignUp
   *
   * @returns {void}
   *
   */
  render() {
    return (
      <div className="body grey lighten-5">
        <UserMenu />
        <div className="main">
          <div className="container">
            <div className="addrecipe-form z-depth-4">
              <div className="col l6 m8 s12 offset-l6 offset-m4">
                <h4 className="center">Add Recipe</h4>
                <hr />
              </div>
              <form onSubmit={this.handleOnsubmit}>
                <div className="row">
                  <div className="input-field col s12">
                    <input
                      id="title"
                      name="title"
                      type="text"
                      value={this.state.title}
                      onChange={this.handleChange}
                    />
                    <label htmlFor="title" className="active">Title</label>
                  </div>
                  <span className="right red-text">
                    {this.state.errors.titleError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <input
                      id="description"
                      name="description"
                      type="text"
                      value={this.state.description}
                      onChange={this.handleChange}
                    />
                    <label
                      htmlFor="description"
                      className="active"
                    >Description
                    </label>
                  </div>
                  <span className="right red-text">
                    {this.state.errors.descriptionError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <textarea
                      id="ingredients"
                      name="ingredients"
                      value={this.state.ingredients}
                      onChange={this.handleChange}
                      placeholder="enter ingredients seperated by semicolon(;)"
                    />
                  </div>
                  <span className="right red-text">
                    {this.state.errors.ingredientsError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <textarea
                      id="procedures"
                      name="procedures"
                      value={this.state.password}
                      onChange={this.handleChange}
                      placeholder="enter procedures seperated by semicolon(;)"
                    />
                  </div>
                  <span className="right red-text">
                    {this.state.errors.proceduresError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <i className="material-icons prefix">insert_photo</i>
                    <input
                      type="file"
                      name="profileImage"
                      onChange={e => this.handleImageChange(e)}
                    />
                  </div>
                  <span className="right red-text">
                    {this.state.errors.imageError}
                  </span>
                </div>
                <div className="row">
                  <button
                    className="btn right green"
                    type="submit"
                    name="action"
                  >Post
                  </button>
                </div>
                <hr />
                <div className="row center">
                  <div>
                    {this.state.disabled ?
                      <SmallPreloader />
                    :
                    ''
                    }
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

AddRecipe.propTypes = {
  createRecipe: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired,
  recipeReducer: PropTypes.shape({
    isCreated: PropTypes.bool.isRequired
  }).isRequired
};

const mapStateToProps = state => ({
  recipeReducer: state.recipeReducer,
});

export default connect(mapStateToProps, { createRecipe })(AddRecipe);

