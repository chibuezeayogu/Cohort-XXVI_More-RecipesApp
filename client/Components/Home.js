import React, { Component } from 'react';
import { Row, Input, Icon, Textarea, Slider, Slide, Parallax } from 'react-materialize';
import NavBar from  './NavBar';
import Footer from  './Footer';


class Home extends Component {
    render() {
        return (
            <div>
                <Slider className="full-screen slide-margin-top">
                    <Slide
                        src="https://static.pexels.com/photos/161640/abstract-barbecue-barbeque-bbq-161640.jpeg"
                        title="More Recipes">
                        More-Recipes provides a platform for users to share the awesome and exciting recipe ideas they have invented or learnt.
                    </Slide>
                    <Slide
                        src="https://static.pexels.com/photos/213621/pexels-photo-213621.jpeg"
                        title="Easy to Setup"
                        placement="left">
                        npm install react-materialize.
                    </Slide>
                    <Slide
                        src="https://static.pexels.com/photos/51423/bun-chocolate-sweets-dessert-51423.jpeg"
                        title="Unlimited Oppurtunities"
                        placement="right">
                        Contribute and Help to Grow this Project.
                    </Slide>
                </Slider>
            </div>
        );
    }
}

export default Home;