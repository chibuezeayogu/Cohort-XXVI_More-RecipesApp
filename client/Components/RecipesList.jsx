import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Pagination from 'rc-pagination';
import { connect } from 'react-redux';
import { getAllRecipes } from '../action/actionCreators';
import RecipeCard from './RecipeCard.jsx';
import UserMenu from './Header/UserMenu.jsx';
import Footer from './Footer/Footer.jsx';
import Preloader from './Preloder.jsx';
import 'rc-pagination/assets/index.css';

/**
 *
 * @description RecipesList used to display recipes
 *
 * @class
 *
 * @extends Component
 *
 */
class RecipesList extends Component {
  constructor() {
    super();
    this.onChange = this.onChange.bind(this);
    this.state = ({ isLoading: true });
  }

  /**
   * @description fetch all recipe before component mounts
   *
   * @method
   *
   * @memberOf RecipesList
   *
   * @returns {void}
   *
   */
  componentWillMount() {
    const token = localStorage.getItem('jwtToken');
    if (!token) {
      this.props.history.push('/');
    } else {
      const currentPage = this.props.location.search.substring(6);
      if (Number(currentPage) && Number(currentPage) > 0) {
        const offset = 8 * (currentPage - 1);
        this.props.getAllRecipes(offset);
      } else {
        this.props.getAllRecipes(0);
      }
    }
  }

   /**
   * @description checks if next recipes is fetched and disables is loading
   *
   * @method
   *
   * @memberOf RecipesList
   *
   * @param {Object} nextProps - nextProps object
   *
   * @returns {void}
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.recipeReducer.isFetched) {
      this.setState({ isLoading: false });
    }
  }

  /**
   * @description handels page change
   *
   * @method
   *
   * @memberOf RecipesList
   *
   * @param {page} page - current page
   *
   * @returns {void}
   */
  onChange(page) {
    this.props.history.push(`/recipes?page=${page}`);
    const offset = 8 * (page - 1);
    this.props.getAllRecipes(offset);
  }

  /**
   *
   * @description renders JSX element
   *
   * @method
   *
   * @memberOf RecipeList
   *
   * @returns {void}
   *
   */
  render() {
    const { recipes, pagination } = this.props.recipeReducer;
    let allRecipes, renderPagination;
    if (recipes && recipes.length === 0) {
        allRecipes = <h4 className="center-align">This page has no recipe</h4>;
    } else if (recipes.length > 0) {
      allRecipes = recipes.map(recipe => (<RecipeCard
        {...this.props}
        key={recipe.id}
        id={recipe.id}
        recipe={recipe}
      />));
    }
    return (
      <div className="body grey lighten-5">
        <UserMenu />
        <div className="main">
          <div className="container">
            <div className="row">
              <h4 className="center">Recipes</h4>
              <hr />
            </div>
            <div className="row left align-recipe" style={{ width: '100%' }}>
              {this.state.isLoading ? <Preloader /> : allRecipes}
            </div>
          </div>
        </div>
        <div className="row s12 m6 l3">
        { recipes.length > 0 ?
            <Pagination
              onChange={this.onChange}
              current={pagination.currentPage}
              pageSize={pagination.pageSize}
              total={pagination.totalCount}
            />
          : ''
          }
        </div>
        <Footer />
      </div>
    );
  }
}

RecipesList.propTypes = {
  recipeReducer: PropTypes.shape({
    pagination: PropTypes.shape({
      currentPage: PropTypes.number,
      pageSize: PropTypes.number,
      totalCount: PropTypes.number
    }),
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired,
  location: PropTypes.shape({
    search: PropTypes.string
  }).isRequired,
  getAllRecipes: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  recipeReducer: state.recipeReducer,
  userData: state.userData,
  favouriteReducer: state.favouriteReducer
});

export default connect(mapStateToProps, { getAllRecipes })(RecipesList);
