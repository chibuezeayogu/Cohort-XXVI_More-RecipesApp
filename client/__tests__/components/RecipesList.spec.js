import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import { RecipesList } from '../../Components/RecipesList';
import mockData from '../__mock__/actionMockData';

const props = {
  recipeReducer: {
    recipes: [],
    pagination: {},
    isFetched: false
  },
  history: {
    push: jest.fn(),
  },
  location: {
    search: '?page=1',
  },
  fetchAllRecipes: jest.fn(),
};

describe('<RecipesList />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<RecipesList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<RecipesList />', () => {
  it('should call componentWillReceiveProps', () => {
    const wrapper = shallow(<RecipesList {...props} />);
    const componentWillReceivePropsSpy = jest
      .spyOn(wrapper.instance(), 'componentWillReceiveProps');
    wrapper.setProps({
      ...props,
      recipeReducer:
      {
        isFetched: true,
        recipes: mockData.data.recipes,
        pagination: mockData.data.pagination
      }
    });
    wrapper.setState({ isLoading: false });
    expect(componentWillReceivePropsSpy).toHaveBeenCalled();
  });
});

describe('<RecipesList />', () => {
  it('should call onChanged function', () => {
    const wrapper = shallow(<RecipesList {...props} />);
    const onChangeSpy = jest.spyOn(wrapper.instance(), 'onChange');

    wrapper.instance().onChange();
    expect(onChangeSpy).toHaveBeenCalled();
    expect(wrapper.instance().props.fetchAllRecipes).toHaveBeenCalled();
    expect(wrapper.instance().props.history.push).toHaveBeenCalled();
  });
});
