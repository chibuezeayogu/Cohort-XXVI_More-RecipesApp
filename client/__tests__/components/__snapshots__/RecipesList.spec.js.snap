// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipesList /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  "node": <div
    className="body grey lighten-5"
  >
    <withRouter(Connect(UserMenu))
      fetchAllRecipes={
        [MockFunction] {
          "calls": Array [
            Array [
              0,
            ],
          ],
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      location={
        Object {
          "search": "?page=1",
        }
      }
      recipeReducer={
        Object {
          "isFetched": false,
          "pagination": Object {},
          "recipes": Array [],
        }
      }
    />
    <div
      className="main"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <h4
            className="center"
          >
            Recipes
          </h4>
          <hr />
        </div>
        <div
          className="row left align-recipe"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <Preloader />
        </div>
      </div>
    </div>
    <div
      className="row s12 m6 l3"
    >
      
    </div>
    <Footer />
  </div>,
  "nodes": Array [
    <div
      className="body grey lighten-5"
    >
      <withRouter(Connect(UserMenu))
        fetchAllRecipes={
          [MockFunction] {
            "calls": Array [
              Array [
                0,
              ],
            ],
          }
        }
        history={
          Object {
            "push": [MockFunction],
          }
        }
        location={
          Object {
            "search": "?page=1",
          }
        }
        recipeReducer={
          Object {
            "isFetched": false,
            "pagination": Object {},
            "recipes": Array [],
          }
        }
      />
      <div
        className="main"
      >
        <div
          className="container"
        >
          <div
            className="row"
          >
            <h4
              className="center"
            >
              Recipes
            </h4>
            <hr />
          </div>
          <div
            className="row left align-recipe"
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <Preloader />
          </div>
        </div>
      </div>
      <div
        className="row s12 m6 l3"
      >
        
      </div>
      <Footer />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <RecipesList
        fetchAllRecipes={
          [MockFunction] {
            "calls": Array [
              Array [
                0,
              ],
            ],
          }
        }
        history={
          Object {
            "push": [MockFunction],
          }
        }
        location={
          Object {
            "search": "?page=1",
          }
        }
        recipeReducer={
          Object {
            "isFetched": false,
            "pagination": Object {},
            "recipes": Array [],
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": RecipesList {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onChange": [Function],
        "props": Object {
          "fetchAllRecipes": [MockFunction] {
            "calls": Array [
              Array [
                0,
              ],
            ],
          },
          "history": Object {
            "push": [MockFunction],
          },
          "location": Object {
            "search": "?page=1",
          },
          "recipeReducer": Object {
            "isFetched": false,
            "pagination": Object {},
            "recipes": Array [],
          },
        },
        "refs": Object {},
        "state": Object {
          "isLoading": true,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="body grey lighten-5"
        >
          <withRouter(Connect(UserMenu))
            fetchAllRecipes={
              [MockFunction] {
                "calls": Array [
                  Array [
                    0,
                  ],
                ],
              }
            }
            history={
              Object {
                "push": [MockFunction],
              }
            }
            location={
              Object {
                "search": "?page=1",
              }
            }
            recipeReducer={
              Object {
                "isFetched": false,
                "pagination": Object {},
                "recipes": Array [],
              }
            }
          />
          <div
            className="main"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <h4
                  className="center"
                >
                  Recipes
                </h4>
                <hr />
              </div>
              <div
                className="row left align-recipe"
                style={
                  Object {
                    "width": "100%",
                  }
                }
              >
                <Preloader />
              </div>
            </div>
          </div>
          <div
            className="row s12 m6 l3"
          >
            
          </div>
          <Footer />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="body grey lighten-5"
        >
          <withRouter(Connect(UserMenu))
            fetchAllRecipes={
              [MockFunction] {
                "calls": Array [
                  Array [
                    0,
                  ],
                ],
              }
            }
            history={
              Object {
                "push": [MockFunction],
              }
            }
            location={
              Object {
                "search": "?page=1",
              }
            }
            recipeReducer={
              Object {
                "isFetched": false,
                "pagination": Object {},
                "recipes": Array [],
              }
            }
          />
          <div
            className="main"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <h4
                  className="center"
                >
                  Recipes
                </h4>
                <hr />
              </div>
              <div
                className="row left align-recipe"
                style={
                  Object {
                    "width": "100%",
                  }
                }
              >
                <Preloader />
              </div>
            </div>
          </div>
          <div
            className="row s12 m6 l3"
          >
            
          </div>
          <Footer />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RecipesList
    fetchAllRecipes={
      [MockFunction] {
        "calls": Array [
          Array [
            0,
          ],
        ],
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={
      Object {
        "search": "?page=1",
      }
    }
    recipeReducer={
      Object {
        "isFetched": false,
        "pagination": Object {},
        "recipes": Array [],
      }
    }
  />,
}
`;
